src/algorithm/iter.rs

```diff
@@ -0,0 +1,244 @@
+///  first child of the parent(as returned by hugr.children()).
+pub fn recursive_children_postorder(
+    hugr: &impl HugrView,
+    root: Node,
+) -> impl '_ + Iterator<Item = Node> {
+    recursive_children(hugr, root).filter_map(VisitOrder::into_postorder)
+}
+
+#[cfg(test)]
+mod test {
+    use std::error::Error;
+
+    use rstest::rstest;
+
+    use crate::{
+        builder::{
+            test::simple_dfg_hugr, Dataflow, DataflowHugr, DataflowSubContainer, FunctionBuilder,
+        },
+        extension::prelude::USIZE_T,
+        hugr::hugrmut::sealed::HugrMutInternals,
+        ops,
+        ops::OpType,
+        type_row,
+        types::FunctionType,
+    };
+
+    use super::*;
+
+    fn node_string(hugr: &impl HugrView, node: Node) -> &'static str {
+        match *hugr.get_optype(node) {
+            OpType::FuncDefn(_) => "F",
+            OpType::DFG(_) => "D",
+            OpType::Input(_) => "I",
+            OpType::Output(_) => "O",
+            OpType::LeafOp(ops::LeafOp::Noop { .. }) => "N",
+            _ => "?",
+        }
+    }
+
+    fn visit_nodeorder_string(hugr: &impl HugrView, node: VisitOrder<Node>) -> String {
+        match node {
+            VisitOrder::PostOrder(x) => node_string(hugr, x).to_string(),
+            VisitOrder::PreOrder(x) => format!("Pre({})", node_string(hugr, x)),
+        }
+    }
+
+    fn visitorder_string(
+        hugr: &impl HugrView,
+        iter: impl IntoIterator<Item = VisitOrder<Node>>,
+    ) -> String {
+        let mut s = String::new();
+        for x in iter {
+            s += &visit_nodeorder_string(hugr, x)
+        }
+        s
+    }
+
+    fn visitnode_string(hugr: &impl HugrView, iter: impl IntoIterator<Item = Node>) -> String {
+        let mut s = String::new();
+        for x in iter {
+            s += node_string(hugr, x)
+        }
+        s
+    }
+
+    #[rstest]
+    fn simple(simple_dfg_hugr: crate::Hugr) {
+        assert_eq!(
+            visitorder_string(
+                &simple_dfg_hugr,
+                super::recursive_children(&simple_dfg_hugr, simple_dfg_hugr.root())
+            ),
+            "Pre(D)IOD"
+        );
+        assert_eq!(
+            visitnode_string(
+                &simple_dfg_hugr,
+                super::recursive_children_postorder(&simple_dfg_hugr, simple_dfg_hugr.root())
+            ),
+            "IOD"
+        );
+    }
+
+    #[test]
+    fn out_of_order() -> Result<(), Box<dyn Error>> {
+        use crate::ops::handle::NodeHandle;
+        let sig = crate::types::PolyFuncType::new(
+            [],
+            FunctionType::new(type_row![USIZE_T], type_row![USIZE_T]),
+        );
+        let mut fun_builder = FunctionBuilder::new("f3", sig.clone())?;
+        let [i] = fun_builder.input_wires_arr();
+        let noop = fun_builder.add_dataflow_op(ops::LeafOp::Noop { ty: USIZE_T }, [i])?;
+        let dfg_builder = fun_builder.dfg_builder(sig.body().clone(), None, [noop.out_wire(0)])?;
```

Comment:
can use functiontype directly
```suggestion
        let sig = FunctionType::new_endo(type_row![USIZE_T]);
        let mut fun_builder = FunctionBuilder::new("f3", sig.clone().into())?;
        let [i] = fun_builder.input_wires_arr();
        let noop = fun_builder.add_dataflow_op(ops::LeafOp::Noop { ty: USIZE_T }, [i])?;
        let dfg_builder = fun_builder.dfg_builder(sig.clone(), None, [noop.out_wire(0)])?;
```